tests/src/test/js/it/scenarios/isolated_users/list.js

```diff
@@ -0,0 +1,160 @@
+import chai, { describe } from "https://jslib.k6.io/k6chaijs/4.3.4.2/index.js";
+import {
+  authenticateWeb,
+  createEmptyStructure
+} from "https://raw.githubusercontent.com/edificeio/edifice-k6-commons/develop/dist/index.js";
+import {
+  listIsolated,
+  createUser,
+  detachUserFromStructures,
+  deleteOrPresuppressUsers
+} from "../admin/utils.js";
+
+chai.config.logFailures = true;
+
+export const options = {
+  setupTimeout: "1h",
+  maxRedirects: 0,
+  thresholds: {
+    checks: ["rate == 1.00"],
+  },
+  scenarios: {
+    listIsolated: {
+      exec: 'testListIsolated',
+      executor: "per-vu-iterations",
+      vus: 1,
+      maxDuration: "5s",
+      gracefulStop: '5s',
+    },
+  },
+};
+
+
+function filterById(results, ids) {
+  return results.filter( elem => ids.filter(id=>id===elem.id).length>0 );
+}
+
+/**
+ * @returns A test dataset containing
+ *  - admc: an ADMC user
+ *  - users: a list of isolated users, without structure attachment
+ */
+export function setup() {
+  let data;
+  describe("[Isolated users] Initialize data", () => {
+    const schoolName = `IT Isolated users`;
+    const session = authenticateWeb(__ENV.ADMC_LOGIN, __ENV.ADMC_PASSWORD);
+    //////////////////////////////////
+    // Create 1 structure and 4 users
+    const structure = createEmptyStructure(schoolName, false, session);
+    const user1 = createUser(structure, "User1", "Isolated", "Personnel", session);
+    detachUserFromStructures(user1, [structure], session);
+    const user2 = createUser(structure, "User2", "Isolated", "Teacher", session);
+    detachUserFromStructures(user2, [structure], session);
+    const user3 = createUser(structure, "User3", "Isolated", "Personnel", session);
+    detachUserFromStructures(user3, [structure], session);
+    const user4 = createUser(structure, "User4", "Isolated", "Relative", session);
+    detachUserFromStructures(user4, [structure], session);
+
+    data = {users: [user1, user2, user3, user4]};
+  });
+  return data;
+}
+/** Cleanup isolated users. */
+export function teardown({users}) {
+  const session = authenticateWeb(__ENV.ADMC_LOGIN, __ENV.ADMC_PASSWORD);
+  if( users && session )
+    deleteOrPresuppressUsers(users, session);
+}
+
+/**
+ * Ensure that :
+ * - isolated users are found
+ * - isolated users are found in correct sort order
+ * - code-injection is rejected when listing isolated users
+ * @param {*} param0 Initialized data
+ */
+export function testListIsolated({users}) {
+  describe("[Isolated users] List basic", () => {
+    const session = authenticateWeb(__ENV.ADMC_LOGIN, __ENV.ADMC_PASSWORD);
+    tryListing(users, session);
+    tryListingInDescOrder(session);
+    tryInjectCode(session);
+  });
+};
+
+function tryListing(users, session) {
+  const requesterType = "ADMC";
+  const userIds = users.map(u=>u.id);
+  const results = listIsolated(session);
+  const checks = {};
+  checks[`${requesterType} should be able to list 4 isolated users`] = () => {
```

Comment:
Tu peux enlever le 4 du nom de la vérification parce qu'en soi la valeur ne porte pas de sens. `xxx should be able to list all isolated users` c'est plutôt pas mal